pkg/osquery/tables/screenlock/screenlock.go

```diff
@@ -0,0 +1,123 @@
+// +build darwin
+
+// Screenlock calls out to osquery to get the screenlock status.
+//
+// While this could be implemneted as a
+// `dataflattentable.TablePluginExec` table, instead we have a
+// dedicated table. This allows us to have a consistent set of
+// columns, and change the implementation as desired.
+
+package screenlock
+
+import (
+	"context"
+	"encoding/json"
+	"strings"
+
+	"github.com/kolide/launcher/pkg/cmdwrapper"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+	"github.com/pkg/errors"
+)
+
+type Table struct {
+	client   *osquery.ExtensionManagerClient
+	logger   log.Logger
+	osqueryd string
+}
+
+type osqueryScreenLockRow struct {
+	Enabled     string `json:"enabled"`
+	GracePeriod string `json:"grace_period"`
+}
+
+func TablePlugin(client *osquery.ExtensionManagerClient, logger log.Logger, osqueryd string) *table.Plugin {
+
+	columns := []table.ColumnDefinition{
+		table.TextColumn("user"),
+		table.IntegerColumn("enabled"),
+		table.IntegerColumn("grace_period"),
+	}
+
+	t := &Table{
+		client:   client,
+		logger:   logger,
+		osqueryd: osqueryd,
+	}
+
+	return table.NewPlugin("kolide_screenlock", columns, t.generate)
+
+}
+
+func (t *Table) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+
+	userQ, ok := queryContext.Constraints["user"]
+	if !ok || len(userQ.Constraints) == 0 {
+		return nil, errors.New("The kolide_screenlock table requires a user")
+	}
+
+	for _, userConstraint := range userQ.Constraints {
+		if !onlyAllowedCharacters(userConstraint.Expression) {
+			level.Info(t.logger).Log("msg", "Disallowed character in user expression")
+			return nil, errors.New("Disallowed character in user expression")
+		}
+
+		stdout, stderr, err := cmdwrapper.Run(
+			ctx,
+			t.osqueryd,
+			[]string{
+				"--config_path", "/dev/null",
+				"-S",
+				"--json",
+				"select enabled, grace_period from screenlock",
+			},
+			cmdwrapper.RunAsUser(userConstraint.Expression),
+		)
+
+		if err != nil {
+			level.Info(t.logger).Log(
+				"msg", "Error getting screenlock status",
+				"stderr", stderr,
+				"stdout", stdout,
+				"err", err,
+			)
+			continue
+		}
+
+		var osqueryResults []osqueryScreenLockRow
+
+		if err := json.Unmarshal([]byte(stdout), &osqueryResults); err != nil {
+			level.Info(t.logger).Log(
+				"msg", "error unmarshalling json",
+				"err", err,
+				"stdout", stdout,
+			)
+			continue
+		}
+
+		for _, row := range osqueryResults {
+			res := map[string]string{
+				"user":         userConstraint.Expression,
+				"enabled":      row.Enabled,
+				"grace_period": row.GracePeriod,
+			}
+			results = append(results, res)
+		}
+	}
+	return results, nil
+}
+
+const allowedCharacters = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789_"
```

Comment:
This constraint should include `-` for example my username would throw an error:
```
osquery> SELECT * FROM kolide_screenlock WHERE user = 'fritz-imac';
{"caller":"screenlock.go:65","msg":"Disallowed character in user expression","severity":"info","ts":"2020-06-22T12:13:49.219496Z"}
I0622 08:13:49.219754 435154368 virtual_table.cpp:1008] Invalid response from the extension table. Error 1: error generating table: Disallowed character in user expression
Error: error generating table: Disallowed character in user expression
osquery> 
```

Likewise we should have some consideration for diacritical marks such as: `Björk`

Reply:
Yeah... that's a good point. I'll update that.

Reply:
Actually.... this table really only needs the user id, not the username. Do you think it makes sense to use that instead? Less lookups

Reply:
Because this table only works when run as the current logged_in_user I imagined the ideal scenario was as proposed in the issue, where we join on `logged_in_users` and return only the `console` type user, and there is no output if a user is not currently logged in:
```
"SELECT user AS username, enabled, grace_period FROM logged_in_users, screenlock WHERE tty = 'console';"
```

Reply:
I think one of the things we've concludes from the last year of use, is that launcher tables are better as small building blocks. Logic should be up a level, and conveyed via sql.

Accordingly, I think this should _do less_ and just take user id as a parameter.

Reply:
I'm even skeptical that we want this table, and not something more general. But starting with this seems like an easy trial.

Reply:
After thinking a bunch more, and tinkering with code, I've concluded that I want to do the initial implementation as a requires `user`.  I believe our main fear was getting misleading data. Testing, we appear fine. I get the same sorts of misleading data for all users vs logged in users. See comments in the new PR

I've opted to re-do the PR, using something simpler. I don't think the cmdwrapper is a win at this time.